src/ROOTReader.cc

```diff
@@ -293,42 +297,41 @@ void ROOTReader::openFiles(const std::vector<std::string>& filenames) {
     versionBranch->GetEntry(0);
   }
   m_fileVersion = versionPtr ? *versionPtr : podio::version::Version{0, 0, 0};
-  delete versionPtr;
```

Comment:
This is the only case where there is a `delete` without a `new. It doesn't seem to leak without the `delete` so it seems removing it it's fine, also I don't understand why it should be deleted since there is no `new`. I think it is fine to delete it because the contents get copied to `m_fileVersion` so they will never be used again, although it is not needed to do so.

Reply:
This might be a leftover from a time when there was a `auto versionPtr = new podio::version::Version()`, but I am not sure. I think it actually should do that and it only works at the moment because `sizeof(podio::version::Version)` is smaller than that of a pointer. Or ROOT does some allocation for us (and then maybe also cleans up properly?).

Looking at the logic, it should also be possible to simply use a pointer to `m_fileVersion` and pass that into `versionBranch->SetAddress`, since if the branch is present, then also reading should work and we don't actually need the intermediate pointer to store this into. `m_fileVersion` is initialized to `0,0,0` in any case.

Reply:
Yes, I thought the same when I was going through this part and this change was merged in this PR.